### PR TITLE
Issue 83 replace static classes with namespaces

### LINKED ISSUE #83: Replace static classes with namespaces
There are a few static classes like solver/SolverSelection, boundaryCondition/DomainBoundary, boundaryCondition/ObstacleBoundary, utility/Utility, which also can be namespaces.

### PR DESCRIPTION
**Pull request by ChristianVonMach**
suggestion to close #83
I only rewrote the mentioned classes in the issue.

### CODE REVIEW

**Comment by ChristianVonMach** [comment]
> I don't like, that you cannot restrict the access to private methods in namespaces. Is there another way to prevent the usage of the formerly private methods?

There is and it is done in 2b1251d

File: src/utility/Utility.h [hunk 1]
```diff
@@ -17,29 +17,25 @@
...
-//  static size_t getCoordinateI(size_t idx,
-//                               size_t Nx,
-//                               size_t Ny,
-//                               size_t j,
-//                               size_t k);
-//  static size_t getCoordinateJ(size_t idx, size_t Nx, size_t Ny, size_t k);
-//  static size_t getCoordinateK(size_t idx, size_t Nx, size_t Ny);
-    static std::vector<std::string> split(const char* text, char delimiter);
-    static std::vector<std::string> split(const std::string& text,
+//  size_t getCoordinateI(size_t idx,
```
Comment (LinhWuerzburger) [inline src/utility/Utility.h:24]: just remove the now unnecessary methods getCoordinateI/J/K

**Review by LinhWuerzburger — COMMENTED** [review]
I am not sure, why you changed the order of the methods in boundaryCondition. To me it seems unnecessary and only complicates the review process. Moreover, I prefer to put public methods at the top of the file rather than at the bottom.

The changes from static class to namespace itself are fine the way they are.

File: src/utility/Utility.h (hunk 1 above)
Comment (ChristianVonMach) [reply src/utility/Utility.h:24]: done in a47401d

**Comment by ChristianVonMach** [comment]
> I am not sure, why you changed the order of the methods in boundaryCondition. To me it seems unnecessary and only complicates the review process. Moreover, I prefer to put public methods at the top of the file rather than at the bottom.

this would require a forward declaration of apply_{dirichlet, neumann, ...}. I think a topological sort of functions makes more sense in C/C++.

**Review by LinhWuerzburger — APPROVED** [review]

**Pull request merged**